### PR TITLE
 Add support for AWS Direct Connect Sitelink #22187

### LINKED ISSUE #22187: Add support for AWS Direct Connect Sitelink
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Added [DX Sitelink support](https://aws.amazon.com/blogs/networking-and-content-delivery/introducing-aws-direct-connect-sitelink/) which requires updates to DX VIF resources.

### New or Affected Resource(s)

* aws_dx_hosted_private_virtual_interface
* aws_dx_hosted_public_virtual_interface
* aws_dx_hosted_transit_virtual_interface
* aws_dx_private_virtual_interface
* aws_dx_public_virtual_interface
* aws_dx_transit_virtual_interface

### Potential Terraform Configuration

```hcl
resource "aws_dx_private_virtual_interface" "foo" {
  connection_id = "dxcon-zzzzzzzz"

  name            = "vif-foo"
  vlan            = 4094
  address_family  = "ipv4"
  bgp_asn         = 65352
  enable_sitelink = true
}
```

### PR DESCRIPTION
**Pull request by NetDevAutomate**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #22187.

Output from acceptance testing:

```
$ make testacc TESTS=TestAccXXX PKG=ec2

...
```

### CODE REVIEW

**Review by drewmullen — CHANGES_REQUESTED** [review]
great work! looking forward to seeing the tests

Can you also please update the docs which are located in `website/docs/{r,d}/`

When youre ready can you also please create a [change log file](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#pull-request-types-to-changelog): `.changelog/22350.txt`

**Comment by NetDevAutomate** [comment]
> great work! looking forward to seeing the tests
> 
> Can you also please update the docs which are located in `website/docs/{r,d}/`
> 
> When youre ready can you also please create a [change log file](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#pull-request-types-to-changelog): `.changelog/22350.txt`

Website docs (r) modified and changelog file created. Many thanks

File: internal/service/directconnect/vif.go [hunk 1]
```diff
@@ -28,12 +28,22 @@ func dxVirtualInterfaceRead(id string, conn *directconnect.DirectConnect) (*dire
 func dxVirtualInterfaceUpdate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*conns.AWSClient).DirectConnectConn
 
-	if d.HasChange("mtu") {
+	if d.HasChanges("mtu") {
```
Comment (ewbankkit) [inline internal/service/directconnect/vif.go:31]: If you are making separate calls to [`UpdateVirtualInterfaceAttributes`](https://docs.aws.amazon.com/directconnect/latest/APIReference/API_UpdateVirtualInterfaceAttributes.html) then please keep `d.HasChange()`.

**Comment by ewbankkit** [comment]
@drewmullen Would you be able to verify acceptance tests for this?

**Comment by NetDevAutomate** [comment]
> @ ewbankkit  Many thanks, I have corrected this.

Many thanks for spotting this, much appreciated.

**Comment by drewmullen** [comment]
> @drewmullen Would you be able to verify acceptance tests for this?

There is a unique quota requirement for the service and I believe Andy is still waiting for the update to his account before he can run the tests.  We'll make sure to add the quota update to the CI accounts as well.

**Comment by NetDevAutomate** [comment]
> 

I hope to get my account permissions sorted by Thursday 6th Jan and can then run the final tests. Manual testing has not shown any issues, so all being well, I'll update with positive results this week.

**Comment by ewbankkit** [comment]
@NetDevAutomate Thanks. Apart from the `terrafmt` linter error, LGTM.

**Comment by ewbankkit** [comment]
Will also need a [CHANGELOG entry](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#changelog-process).

**Comment by NetDevAutomate** [comment]
> Will also need a [CHANGELOG entry](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#changelog-process).

Many thanks again, changelog entry created and formatting issue corrected.

**Comment by NetDevAutomate** [comment]
$ make testacc TESTARGS='-run=TestAccTransitVirtualInterfaceSiteLink_basic' ACCTEST_PARALLELISM=5 PKG=directconnect
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/directconnect/... -v -count 1 -parallel 5 -run=TestAccTransitVirtualInterfaceSiteLink_basic -timeout 180m
=== RUN   TestAccTransitVirtualInterfaceSiteLink_basic
--- PASS: TestAccTransitVirtualInterfaceSiteLink_basic (964.55s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/directconnect	969.526s

$ make testacc TESTARGS='-run=TestAccDirectConnectPrivateVirtualInterfaceSiteLink_basic' ACCTEST_PARALLELISM=5 PKG=directconnect
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/directconnect/... -v -count 1 -parallel 5 -run=TestAccDirectConnectPrivateVirtualInterfaceSiteLink_basic -timeout 180m
=== RUN   TestAccDirectConnectPrivateVirtualInterfaceSiteLink_basic
=== PAUSE TestAccDirectConnectPrivateVirtualInterfaceSiteLink_basic
=== CONT  TestAccDirectConnectPrivateVirtualInterfaceSiteLink_basic
--- PASS: TestAccDirectConnectPrivateVirtualInterfaceSiteLink_basic (947.63s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/directconnect	952.053s

**Comment by NetDevAutomate** [comment]
$ make testacc TESTARGS='-run=TestAccDirectConnectTransitVirtualInterface_serial' ACCTEST_PARALLELISM=5 PKG=directconnect
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/directconnect/... -v -count 1 -parallel 5 -run=TestAccDirectConnectTransitVirtualInterface_serial -timeout 180m
=== RUN   TestAccDirectConnectTransitVirtualInterface_serial
=== RUN   TestAccDirectConnectTransitVirtualInterface_serial/basic
=== RUN   TestAccDirectConnectTransitVirtualInterface_serial/tags
=== RUN   TestAccDirectConnectTransitVirtualInterface_serial/sitelink
--- PASS: TestAccDirectConnectTransitVirtualInterface_serial (2713.20s)
    --- PASS: TestAccDirectConnectTransitVirtualInterface_serial/basic (1049.07s)
    --- PASS: TestAccDirectConnectTransitVirtualInterface_serial/tags (671.71s)
    --- PASS: TestAccDirectConnectTransitVirtualInterface_serial/sitelink (992.43s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/directconnect	2716.781s

**Comment by ewbankkit** [comment]
@NetDevAutomate @drewmullen Once this goes green are you OK for me to merge? Thanks.

**Comment by NetDevAutomate** [comment]
> @NetDevAutomate @drewmullen Once this goes green are you OK for me to merge? Thanks.

That would be great, many thanks.

**Pull request merged**